shared/oae/js/jquery-plugins/jquery.switchtab.js

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/**
+ * Utility plugin that handles the "switch tab" component interactions.
+ */
+
+define(['jquery', 'jquery.history'], function (jQuery, oaeUtil) {
+    (function($) {
+
+        /**
+         * OAE plugin that implements the interaction of the "Switch Tab" component. A Switch Tab
+         * is essentially a set of tabs that look like a left-to-right switch. There can only be
+         * one selected at one time. Back functionality is also provided.
+         *
+         * Usage:
+         *
+         *  You should create a template that looks like the following:
+         *
+         *  ```
+         *  <div id="search-scope" class="oae-switchtab-container">
```

Comment:
Rather than rolling our own tab switch component entirely, it would be good if we could leverage the existing Bootstrap pill tabs as much as possible. It is already documented, is semantically a bit better and implements a number of accessibility considerations. 

The `data-switchtab-path` part can probably just be replaced by using the native `href` on the standard tabs component.

We'll probably still want this file and component to bind to and act on some of the events that are sent out by the Bootstrap tabs component.